Allow Verifier to validate varchar as floating point (#23312)
prestodb/presto: presto-docs/src/main/sphinx/admin/verifier.rst

    * Checks the nullity of control sum and test sum.
    * If either control mean or test mean very close 0, checks if both are close to 0.
    * Checks the relative error between control sum and test sum.
* **VARCHAR Columns**
    * For ``VARCHAR`` columns, a simple checksum column is generated for verification using :func:`!checksum`.

>     * If ``validate-string-as-double`` is set, 6 additional columns below are generated. If ``NULL`` counts

Reviewer: ```suggestion
    * If ``validate-string-as-double`` is set, these six columns are generated. If ``NULL`` counts
```

>       are equal before and after casting all values to ``DOUBLE``, apply the aforementioned floating point validation.

Reviewer: ```suggestion
      are equal before and after casting all values to ``DOUBLE``, apply the floating point validation.
```

      Otherwise, check if the simple checksum matches.

>         * Count of the ``NULL`` values

Reviewer: Error in local doc build: 

`/Users/steveburnett/Documents/GitHub/presto/presto-docs/src/main/sphinx/admin/verifier.rst:174: ERROR: Unexpected indentation.`

Suggest unindenting this list, and any nested list within it, and do a local doc build until these similar errors do not appear. See [Building the Documentation](https://github.com/prestodb/presto/tree/master/presto-docs#building-the-documentation) for more information. 

Screenshot of local doc build, showing a vertical gray line in the generated file. 
<img width="799" alt="Screenshot 2024-08-05 at 11 27 58 AM" src="https://github.com/user-attachments/assets/560984ab-b331-40dc-ae02-df7d0474a372">

Author: @steveburnett , thanks a lot for pointing to the local build. That's very convenient! I've fixed the build errors.

        * Count of the ``NULL`` values after all values are casted to ``DOUBLE``

>         * After casting all values to ``DOUBLE``

Reviewer: ```suggestion
        After casting all values to ``DOUBLE``
```
Suggest not formatting this line as an item in the list, to help the reader by showing them six items instead of the formatting suggesting there are seven.

            * Sum of the finite values
            * ``NAN`` count of the values
            * Positive infinity count of the values
            * Negative infinity count of the values
* **Array Columns**
    * For an array column ``arr`` of type ``array(E)``, 3 columns are generated for verification:
        * Sum of the cardinality
        * Checksum of the cardinality

>         * Array checksum

Reviewer: ```suggestion
        Array checksum
```
Suggest not formatting this line as an item in the list, to help the reader. The two sub-items formatting is good, as your description clearly tells the reader the next two are an either/or.

Author: Need to keep this as one item in the list. "Array checksum" is one of the three columns generated.

            * If ``E`` is not orderable, array checksum is ``checksum(arr)``.
            * If ``E`` is orderable, array checksum is ``coalesce(checksum(try(array_sort(arr))), checksum(arr))``.
    * If ``use-error-margin-for-floating-point-arrays`` is set and E is ``DOUBLE`` or ``REAL``,

>       6 columns below are generated instead. Check if the sum of the cardinality matches and if the checksum of the cardinality

Reviewer: ```suggestion
      these six columns are generated instead. Check if the sum of the cardinality matches and if the checksum of the cardinality
```

>       matches. And apply the aforementioned floating point validation to the rest of results.

Reviewer: ```suggestion
      matches. Apply the aforementioned floating point validation to the rest of results.
```

>         * Sum of the cardinality

Reviewer: Error in local doc build: 

`/Users/steveburnett/Documents/GitHub/presto/presto-docs/src/main/sphinx/admin/verifier.rst:191: ERROR: Unexpected indentation.`

Suggest unindenting this list, and any nested list within it, and do a local doc build until these similar errors do not appear. See [Building the Documentation](https://github.com/prestodb/presto/tree/master/presto-docs#building-the-documentation) for more information.

        * Checksum of the cardinality
        * Sum of the finite elements of all array values
        * Count of the ``NAN`` elements of all array values
        * Count of the positive infinity elements of all array values
        * Count of the negative infinity elements of all array values
    * If ``validate-string-as-double`` is set and E is ``VARCHAR``, 9 columns below are generated instead.
      Check if the sum and the checksum of the cardinality match. If ``NULL`` counts are equal before and after
      casting all array elements to ``DOUBLE``, apply the aforementioned floating point validation. Otherwise, check
      if the array checksum matches.

>         * Sum of the cardinality

Reviewer: Error in local doc build: 

`/Users/steveburnett/Documents/GitHub/presto/presto-docs/src/main/sphinx/admin/verifier.rst:201: ERROR: Unexpected indentation.`

Suggest unindenting this list, and any nested list within it, and do a local doc build until these similar errors do not appear. See [Building the Documentation](https://github.com/prestodb/presto/tree/master/presto-docs#building-the-documentation) for more information.

        * Checksum of the cardinality
        * Array checksum ``checksum(array_sort(arr))``
        * Count of the ``NULL`` elements of all array values
        * Count of the ``NULL`` elements after all array elements are casted to ``DOUBLE``

>         * After casting all array elements to ``DOUBLE``

Reviewer: ```suggestion
        After casting all array elements to ``DOUBLE``
```
Suggest not formatting this line as an item in the list, to help the reader.

            * Sum of the finite elements of all array values
            * Count of the ``NAN`` elements of all array values
            * Count of the positive infinity elements of all array values
            * Count of the negative infinity elements of all array values
* **Map Columns**
    * For a map column of type ``map(K, V)``, 4 columns are generated for verification:
        * Sum of the cardinality
        * Checksum of the map
        * Array checksum of the key set
        * Array checksum of the value set
    * If ``validate-string-as-double`` is set and K is ``VARCHAR``, 6 additional columns are generated:
        * Count of the ``NULL`` elements of all key sets
        * Count of the ``NULL`` elements of the key sets after all map keys are casted to ``DOUBLE``

>         * After casting all map keys to ``DOUBLE``

Reviewer: ```suggestion
        After casting all map keys to ``DOUBLE``
```
Suggest not formatting this line as an item in the list, to help the reader.

            * Sum of the finite elements of all key sets
            * Count of the ``NAN`` elements of all key sets
            * Count of the positive infinity elements of all key sets
            * Count of the negative infinity elements of all key sets

>     * Additionally if ``validate-string-as-double`` is set and V is ``VARCHAR``, 6 additional columns are generated:

Reviewer: ```suggestion
    * If ``validate-string-as-double`` is set and V is ``VARCHAR``, 6 additional columns are generated:
```

        * Count of the ``NULL`` elements of all value sets
        * Count of the ``NULL`` elements of the value sets after all map values are casted to ``DOUBLE``

>         * After casting all map values to ``DOUBLE``

Reviewer: ```suggestion
        After casting all map values to ``DOUBLE``
```
Suggest not formatting this line as an item in the list, to help the reader.
